### PR TITLE
feat(auth): adds github auth library

### PR DESCRIPTION
**Pull request by scefali**
This PR is part of the effort in moving dependencies into sentry from stand-alone repos. Evan and I decided to make the path of this to be: `sentry/auth/providers` to match the other auth providers. The only real change I made was stripping out the check for Django < 1.8.

Must be merged with: https://github.com/getsentry/getsentry/pull/3431

### CODE REVIEW

File: src/sentry/auth/providers/github/views.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        if user.get("email"):
+            return helper.next_step()
+
+        form = ConfirmEmailForm(request.POST or None)
+        if form.is_valid():
+            user["email"] = form.cleaned_data["email"]
+            helper.bind_state("user", user)
+            return helper.next_step()
+
+        return self.respond("github/enter-email.html", {"form": form})
```
Comment (scefali) [inline src/sentry/auth/providers/github/views.py]: Changed the path of the HTML here from `sentry_auth_github` to `github`

File: src/sentry/auth/providers/github/views.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        org_list = self.client.get_org_list(access_token)
+
+        form = SelectOrganizationForm(org_list, request.POST or None)
+        if form.is_valid():
+            org_id = form.cleaned_data["org"]
+            org = [o for o in org_list if org_id == six.text_type(o["id"])][0]
+            helper.bind_state("org", org)
+            return helper.next_step()
+
+        return self.respond("github/select-organization.html", {"form": form, "org_list": org_list})
```
Comment (scefali) [inline src/sentry/auth/providers/github/views.py]: Changed the path of the HTML here from `sentry_auth_github` to `github`

File: src/sentry/auth/providers/github/views.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+
+class GitHubConfigureView(ConfigureView):
+    def dispatch(self, request, organization, auth_provider):
+        return self.render("github/configure.html")
```
Comment (scefali) [inline src/sentry/auth/providers/github/views.py]: Changed the path of the HTML here from `sentry_auth_github` to `github`

**Review by evanpurkhiser — APPROVED** [review]
lgtm

**Pull request merged**